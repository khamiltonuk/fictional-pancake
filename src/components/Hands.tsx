import React from "react";
import styled, { keyframes } from "styled-components";

const Hand = styled.span`
  background: url("https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/hands-sprite.png")
    transparent;
  width: 312px;
  margin-top: -67px;
  height: 135px;
  text-indent: 100%;
  position: absolute;
  display: block;
  overflow: hidden;
  white-space: nowrap;
  animation-duration: 1s;
  animation-fill-mode: both;
  transform-origin: center bottom;
`;

const LeftHand = styled(Hand)<ApiPropsT>`
  left: 0;
  top: 30%;
  animation-name: ${({ playerTwoChoice }) => {
    return playerTwoChoice === "animate" ? `${leftBounce}` : "none";
  }};
  background-image: ${({ playerOneChoice }) => {
    if (playerOneChoice === "spock") {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/spock.jpg')";
    } else if (playerOneChoice === "lizard") {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/lizard.jpg')";
    } else {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/hands-sprite.png')";
    }
  }};
  background-position: ${({ playerOneChoice }) => {
    if (playerOneChoice === "scissor") {
      return "0 0";
    } else if (playerOneChoice === "paper") {
      return "0 -135px;";
    } else {
      return "-312px -135px;";
    }
  }};
`;

const RightHand = styled(Hand)<ApiPropsT>`
  right: 0;
  top: 50%;
  animation-name: ${(props) => {
    return props.playerTwoChoice === "animate" ? `${rightBounce}` : null;
  }};
  background-image: ${(props) => {
    if (props.playerTwoChoice === "spock") {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/spock.jpg')";
    } else if (props.playerTwoChoice === "lizard") {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/lizard.jpg')";
    } else {
      return "url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/hands-sprite.png')";
    }
  }};
  background-position: ${(props) => {
    if (props.playerTwoChoice === "scissor") {
      return "-624px 135px";
    } else if (props.playerTwoChoice === "paper") {
      return "-625px 0";
    } else {
      return "-312px 0;";
    }
  }};
`;

const rightBounce = keyframes`
0%, 20%, 53%, 80%, 100% {
  transition-timing-function: cubic-bezier(0.215, 0.610, 0.355, 1.000);
  transform: translate3d(0,0,0);
}

40%, 43%, 70%, 90%   {
  transition-timing-function: cubic-bezier(0.755, 0.050, 0.855, 0.060);
  transform: translate3d(30px, -30px, 0) rotate(30deg);
}

`;
const leftBounce = keyframes`
0%, 20%, 53%, 80%, 100% {
  transition-timing-function: cubic-bezier(0.215, 0.610, 0.355, 1.000);
  transform: translate3d(0,0,0);
}

40%, 43%, 70%, 90%   {
  transition-timing-function: cubic-bezier(0.755, 0.050, 0.855, 0.060);
  transform: translate3d(-30px, -30px, 0) rotate(-30deg);
}`;

type ApiPropsT = {
  playerOneChoice: string;
  playerTwoChoice: string;
};

export const Hands = (props: ApiPropsT) => {
  const { playerOneChoice, playerTwoChoice } = props;
  return (
    <>
      <LeftHand {...props}>{playerOneChoice}</LeftHand>
      <RightHand {...props}>{playerTwoChoice}</RightHand>
    </>
  );
};
