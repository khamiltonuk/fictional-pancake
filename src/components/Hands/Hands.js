import React from "react";
import styled from "styled-components";

// #left-hand,
// #right-hand {
// 	background: url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/hands-sprite.png') transparent;
// 	width: 312px;
// 	position: absolute;
//   margin-top: -67px;
// 	height: 135px;
//   text-indent: 100%;
//   overflow: hidden;
//   white-space: nowrap;
// }
// #left-hand {
// 	left: 0;
//   top: 30%;
//   background-position: -312px -135px;
// 	&.paper {background-position: 0 -135px; }
// 	&.scissors {background-position: 0 0; }
//   &.spock {background-image: url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/spock.jpg'); }
//   &.lizard {background-image: url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/lizard.jpg'); }
// }
// #right-hand {
// 	right: 0;
//   top: 50%;
// 	background-position: -312px 0;
// 	&.paper {background-position: -625px 0; }
// 	&.scissors {background-position: -624px 135px; }
//   &.spock {background-image: url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/spock.jpg'); }
//   &.lizard {background-image: url('https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/lizard.jpg'); }
// }

const Hand = styled.span`
  background: url("https://raw.githubusercontent.com/khamiltonuk/test-new/master/images/hands-sprite.png")
    transparent;
  width: 312px;
  margin-top: -67px;
  height: 135px;
  text-indent: 100%;
  display: block;
  overflow: hidden;
  white-space: nowrap;
  background-position: ${props =>
    (props.direction = "left" ? "-312px -135px" : "-312px 0")};
`;

const Hands = props => {
  return <Hand {...props} />;
};

export default Hands;
