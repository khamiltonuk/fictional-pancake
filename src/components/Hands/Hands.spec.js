import React from "react";
import { shallow } from "enzyme";
import Hands from "./Hands";
import toJson from "enzyme-to-json";

describe("Hands", () => {
  test("should match the snapshot", () => {
    const wrapper = shallow(<Hands />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
