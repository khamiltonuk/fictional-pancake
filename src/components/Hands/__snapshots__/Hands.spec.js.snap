// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hands should match the snapshot 1`] = `<styled.span />`;
